### DISCUSSION TITLE
Why is the connection to Zed's collab server required for remote development?

### DISCUSSION THREAD

**Discussion by bananu7** [zed-industries/zed#12381] [Feature Requests]
As answered [here](https://github.com/zed-industries/zed/discussions/12272#discussioncomment-9554284), Zed requires phoning home for remote dev to work. The manual also states 

> Currently the two instances connect via Zed's servers, but we intend to build peer to peer communication in the future.

I've also checked FAQ, that states that for (possibly related) collab sessions:

> [proxying] ensures much greater reliability than peer to peer connections.

Why is this required, if the two machines already have an SSH session open between them? If I want to connect via SSH, and that doesn't work, then the remote server won't help. And if it does, there's no reliability concern in the first place.

**Comment by SomeoneToIgnore** [comment]
Mainly due to the possibility to reuse the existing sync code, instead of reinventing the wheel over ssh.
After all, the very same doc you've linked states that 

> Remote Development is in the early stages of development

so please do not consider current state as something set in stone.

**Comment by sayrer** [comment]
Oh shoot, I would switch, but I really need the SSH feature that VSCode has.
